### PR TITLE
Correctly pack Entry/Exit structs for parent as signer, fallthrough no DNSSEC case

### PR DESCRIPTION
**Pull request by sudheesh001**

### CODE REVIEW

File: target.go [hunk 1]
```diff
@@ -163,6 +163,27 @@ func (s *RecursiveResolver) parseQueryFromRequest(r *http.Request) (*dns.Msg, er
 	}
 }
 
+func isSignedByParent(domainName string, rrs []dns.RR) bool {
+	segments := strings.Split(domainName, ".")
```
Comment (iowaguy) [inline target.go]: There's actually a function for this in the DNS lib: https://pkg.go.dev/github.com/miekg/dns?utm_source=godoc#SplitDomainName

Should probably be used as it handles a couple special cases.

File: target.go [hunk 2]
```diff
@@ -163,6 +163,27 @@ func (s *RecursiveResolver) parseQueryFromRequest(r *http.Request) (*dns.Msg, er
...
+	numberSegments := len(segments)
+	// fail fast
+	if numberSegments < 0 {
+		return false
+	}
+	parentDomain := strings.Join(segments[1:], ".")
+	for _, rrRecord := range rrs {
+		switch rrRecord.(type) {
+		case *dns.RRSIG:
+			t := (rrRecord).(*dns.RRSIG)
```
Comment (iowaguy) [inline target.go]: You can actually include this in the type switch. Saves a line of code, and `t` will always have the correct type if you add more `case`s.

```suggestion
		switch t := rrRecord.(type) {
		case *dns.RRSIG:
```

File: target.go [hunk 3]
```diff
@@ -489,10 +530,14 @@ func (s *RecursiveResolver) resolveQueryWithResolver(q *dns.Msg, r resolver) ([]
...
 			log.Println("Failed serializing DNSSEC proofs:", err)
-			return nil, err
+			// Return the non DNSSEC serialized DNS response as fail-over
+			return response.Pack()
 		}
 
 		response.Extra = append(response.Extra, &dnssecProof)
+
+		b, _ := response.Pack()
+		fmt.Printf("Response Size : %v\n", len(b))
```
Comment (iowaguy) [inline target.go]: I don't think `b` ever gets used except in this printf. Is this just here for debugging?

Comment (sudheesh001) [reply target.go]: Good catch, debugging artifact. Thanks for the reivew

**Pull request merged**